fmriprep/config.py

```diff
@@ -683,3 +686,41 @@ def init_spaces(checkpoint=True):
 
     # Make the SpatialReferences object available
     workflow.spaces = spaces
+
+
+def load_previous_config(output_dir, work_dir, participant=None, new_uuid=True):
+    """
+    Search for existing config file, and carry over previous options.
+    If a participant label is specified, the output directory is searched for
+    the most recent config file. If no config is found, the working directory is searched.
+
+    Returns
+    -------
+    success : bool
+        Existing config found and loaded
+
+    """
+    conf = None
+    if participant:
+        # output directory
+        logdir = Path(output_dir) / "fmriprep" / f"sub-{participant[0]}" / "log"
+        for f in logdir.glob(f"**/{CONFIG_FILENAME}"):
+            conf = conf or f
+            if f.stat().st_mtime > conf.stat().st_mtime:
+                conf = f
+
+    if conf is None:
+        # work directory
+        conf = Path(work_dir) / f".{CONFIG_FILENAME}"
+        if not conf.exists():
+            conf = None
+
+    # load existing config
+    if conf is not None:
+        # settings to avoid reusing
+        skip = {} if new_uuid else {"execution": ("run_uuid",)}
```

Comment:
I think this is backwards. My reading is `opts.reports_only -> new_uuid=False -> skip = {"execution": ("run_uuid",)} -> new run_uuid generated`.

Reply:
I think the idea behind the logic here is:

- if `conf` exists, we have found a previous config file with all options (including `run_uuid`)
- when loading this config, skip reusing the previous `run_uuid` and use the newly generated one (so future runs are still treated as unique)

Reply:
The idea here is:

- if `conf` exists, we have found a previous config file with all options (including `run_uuid`)
- when loading this config, skip reusing the previous `run_uuid` and use the newly generated one (so future runs are still treated as unique)